source/agora/script/Signature.d

```diff
@@ -78,7 +82,13 @@ public struct SigPair
     public inout(ubyte)[] opSlice () inout pure nothrow @safe /*@nogc*/
     {
         return this.signature.toBlob()[] ~ ubyte(this.sig_hash)
-            ~ ((sig_hash & SigHash.Single) ? nativeToLittleEndian(this.output_idx).dup : []);
+            ~ (SigPair.usesOutputIdx(this.sig_hash) ? nativeToLittleEndian(this.output_idx).dup : []);
```

Comment:
We probably shouldn't be allocating in an `opSlice`, but that's an improvement for another day